### ISSUE TITLE
cannot set custom smearing without data (Trac #714)

### ISSUE THREAD

**Issue by pkienzle** [SasView/sasview#837]
To simulate a dataset before measurement, it would be useful to be able to set custom pinhole smear or custom slit smear even when there is no data available.  When I try this I get

```text
Traceback (most recent call last):
  File "/Users/pkienzle/Source/sasview-new/src/sas/sascalc/data_util/calcthread.py", line 268, in _run
    self.compute(*args, **kwargs)
ValueError: operands could not be broadcast together with shapes (0,) (49,) 
```

Migrated from http://trac.sasview.org/ticket/714
```json
{
    "status": "closed",
    "changetime": "2016-10-07T14:48:27",
    "_ts": "2016-10-07 14:48:27.021691+00:00",
    "description": "To simulate a dataset before measurement, it would be useful to be able to set custom pinhole smear or custom slit smear even when there is no data available.  When I try this I get\n{{{\nTraceback (most recent call last):\n  File \"/Users/pkienzle/Source/sasview-new/src/sas/sascalc/data_util/calcthread.py\", line 268, in _run\n    self.compute(*args, **kwargs)\nValueError: operands could not be broadcast together with shapes (0,) (49,) \n}}}",
    "reporter": "pkienzle",
    "cc": "",
    "resolution": "fixed",
    "workpackage": "SasView Bug Fixing",
    "time": "2016-10-06T16:00:12",
    "component": "SasView",
    "summary": "cannot set custom smearing without data",
    "priority": "blocker",
    "keywords": "",
    "milestone": "SasView 4.1.0",
    "owner": "mathieu",
    "type": "defect"
}
```

**Comment by butlerpd** [comment]
Trac update at `2016/10/07 01:18:03`: 
* **butler** commented:

verified on my windows -- The log file has a bit longer of a traceback in some cases.  Seems to have been introduced by Mathieu Doucet when adding the rescale functionality.  


```text
2016-10-06 21:12:22,845 ERROR Traceback (most recent call last):
  File "C:\Users\Paul\git\sasview_new\src\sas\sascalc\data_util\calcthread.py", line 268, in _run
    self.compute(*args, **kwargs)
  File "C:\Users\Paul\git\sasview_new\src\sas\sasgui\perspectives\fitting\model_thread.py", line 187, in compute
    / output[first_bin:last_bin+1]
ValueError: operands could not be broadcast together with shapes (0) (50) 
```



* **butler** changed milestone from "SasView Next Release +1" to "SasView 4.1.0"
* **butler** changed owner from "" to "mathieu"
* **butler** changed priority from "minor" to "blocker"
* **butler** changed status from "new" to "assigned"

**Comment by mdoucet** [comment]
Trac update at `2016/10/07 14:48:27`: 
* **mathieu** commented:

In changeset 286c7571e8019b4c28a92d3d497cac7da825076c:

```text
#!CommitTicketReference repository="sasview" revision="286c7571e8019b4c28a92d3d497cac7da825076c"
Make sure we can compute models w/o data. Fixes #837
```


* **mathieu** changed resolution from "" to "fixed"
* **mathieu** changed status from "assigned" to "closed"
